src/hooks/useFetch.tsx

```diff
@@ -0,0 +1,13 @@
+import React, {useEffect, useState} from 'react'
+
+export function useFetch(url: RequestInfo | URL) {
+    const [data, setData] = useState(null)
+
+    useEffect(() => {
+        fetch(url)
+        .then((response: Response) => response.json())
+        .then((data: any) => setData(data));
+      }, [url])
+
+    return {data};
+}
```

Comment:
entiendo cual es el objectivo de este hook. Pero le cambiaria varias cosas. Primero en lugar de la URL es mejor que reciba una funcion que retorne una promesa. Luego el hook llama a esa funcion, de esa forma tienes la logica del la capa de datos abstracted and encapsulated. 

En general no esta mal, pero te recomiendo que le eches un ojo a https://tanstack.com/query/v3/ tienen una implementacion muy parecida a la tuya. Pero como lo digo felicitaciones por el approach

Reply:
Perfecto, muchas gracias por la revisión de todo, eliminaré el console log y  revisaré los puntos que me comentas. 
Thank you for your comments  Lionel. II really appreciate your experience with us.